lib/ansible/modules/system/systemd-modules-load.py

```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2018, Susant Sahani <ssahani@gmail.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['stableinterface'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: systemd-modules-load
+short_description:  systemd-modules-load
+description:
+    - Allows you to  generate systemd-modules-load configuration files.
+version_added: "2.8"
+options:
+    conf_path:
+        description:
+            - Specifies the path where to write the configuration files.
+        default: "/etc/modules-load.d"
+        choices: [ "/etc/modules-load.d", "/run/modules-load.d", "/usr/lib/modules-load.d" ]
+    file_name:
+        description:
+            - This configuration file name where the configurations will be written. Note the file name will be
+              automatically have the extension .conf.
+    modules:
+        description:
+            - Space separated list of module names.
+    state:
+        description:
+            - Whether configuration files should be added or absentd.
+        choices: [ "present", "absent" ]
+author: "Susant Sahani (@ssahani) <ssahani@gmail.com>"
+'''
+
+EXAMPLES = '''
+# Create config file
+- systemd-modules-load:
+       file_name: my_tunnel
+       modules:  ipip sit vti
+       state: present
+
+# Remove config file
+- systemd-modules-load:
+       file_name: ipip
+       state: absent
+'''
+
+RETURN = r'''
+'''
+
+import os
+import tempfile
+from ansible.module_utils.basic import get_platform, AnsibleModule
+
+UNIT_PATH_MODULES_LOAD = '/usr/lib/modules-load.d'
+UNIT_PATH_MODULES_LOAD_SYSTEM = '/etc/modules-load.d'
+UNIT_PATH_MODULES_LOAD_RUN = '/run/modules-load.d'
+
+
+class SystemdModulesLoad(object):
+
+    def __init__(self, module):
+        self.module = module
+        self.args = self.module.params
+        self.conf_path = module.params['conf_path']
+        self.file_name = module.params['file_name']
+        self.modules = module.params['modules']
+        self.state = module.params['state']
+        self.changed = False
+        self.result = {}
+
+    def remove_files(self):
+        paths = [UNIT_PATH_MODULES_LOAD_RUN, UNIT_PATH_MODULES_LOAD_SYSTEM, UNIT_PATH_MODULES_LOAD]
+        rc = False
+
+        list_conf_files = self.file_name.split(' ')
+        for conf_file in list_conf_files:
+            conf_file += '.conf'
+            if os.path.exists(os.path.join(self.conf_path, conf_file)):
```

Comment:
here you can set 'changed' status in `else`

Reply:
should not be it's in if part ? If files exists then we remove it i.e. changed ?